website/addons/base/views.py

```diff
@@ -319,6 +319,11 @@ def addon_view_file(auth, node, node_addon, file_guid, extras):
     )
 
     ret = serialize_node(node, auth, primary=True)
+
+    forbid_edit = (settings.DISK_SAVING_MODE
```

Comment:
For clarity, this can be simplified:

```python
if node_addon.short_name == 'osfstorage' and settings.DISK_SAVING_MODE:
    ret['user']['can_edit'] = False
```